engines/wintermute/base/gfx/opengl/base_surface_opengl3d.cpp

```diff
@@ -20,15 +20,45 @@
  *
  */
 
+#include "common/algorithm.h"
 #include "engines/wintermute/base/gfx/opengl/base_surface_opengl3d.h"
 #include "engines/wintermute/base/gfx/opengl/base_render_opengl3d.h"
 #include "engines/wintermute/base/gfx/base_image.h"
 #include "graphics/transparent_surface.h"
 
 namespace Wintermute {
 
+void applyColorKey(Graphics::Surface &surf, byte ckRed, byte ckGreen, byte ckBlue, bool replaceAlpha) {
+	// this is taken from Graphics::TransparentSurface
+	// only difference is that we set the pixel
+	// color to transparent black, like D3DX,
+	// if it matches the color key
+	for (int y = 0; y < surf.h; y++) {
+		for (int x = 0; x < surf.w; x++) {
+			uint32 pix = ((uint32 *)surf.getPixels())[y * surf.w + x];
+			uint8 r, g, b, a;
+			surf.format.colorToARGB(pix, a, r, g, b);
+			if (r == ckRed && g == ckGreen && b == ckBlue) {
+				a = 0;
+				r = 0;
+				g = 0;
+				b = 0;
+				((uint32 *)surf.getPixels())[y * surf.w + x] = surf.format.ARGBToColor(a, r, g, b);
+			} else if (replaceAlpha) {
+				a = 255;
+				((uint32 *)surf.getPixels())[y * surf.w + x] = surf.format.ARGBToColor(a, r, g, b);
+			}
+		}
+	}
+}
+
 BaseSurfaceOpenGL3D::BaseSurfaceOpenGL3D(BaseGame *game, BaseRenderOpenGL3D *renderer)
-    : BaseSurface(game), tex(nullptr), renderer(renderer), pixelOpReady(false) {
+	: BaseSurface(game), _tex(0), _renderer(renderer), _imageData(nullptr), _texWidth(0), _texHeight(0) {
+	glGenTextures(1, &_tex);
+}
+
+BaseSurfaceOpenGL3D::~BaseSurfaceOpenGL3D() {
```

Comment:
Just to be safe, this should also delete[] _imageData;

Reply:
Done